### PR TITLE
Eval event cleanup

### PR DESCRIPTION
**Pull request by grahamc**
Document the eval_* events, sort some imports and subscriptions.

Note that I'm immediately breaking the documented API in #1145.

### CODE REVIEW

File: t/Hydra/Event/StepFinished.t [hunk 1]
```diff
@@ -7,7 +7,7 @@ my %ctx = test_init();
 require Hydra::Schema;
 require Hydra::Model::DB;
 use Hydra::Event;
-use Hydra::Event::BuildStarted;
+use Hydra::Event::StepFinished;
```
Comment (cole-h) [inline t/Hydra/Event/StepFinished.t:10]: 👀

**Review by cole-h — APPROVED** [review]
Who doesn't love docs

File: t/Hydra/Event/StepFinished.t (hunk 1 above)
Comment (grahamc) [reply t/Hydra/Event/StepFinished.t:10]: :upside_down_face:

File: doc/manual/src/notifications.md [hunk 2]
```diff
@@ -55,6 +55,30 @@ After processing, the row's `notificationspendingsince` column is set to null.
...
+
+### `eval_cached`
+
+* **Payload:** Exactly one value: an opaque trace ID representing this evaluation.
+* **When:** After the evaluator fetches inputs, if none of the inputs changed.
+* **Delivery Semantics:** Ephemeral. `hydra-notify` must be running to react to this event. No record of this event is stored.
+
+### `eval_failed`
+
+* **Payload:** Exactly one value: an opaque trace ID representing this evaluation.
```
Comment (grahamc) [inline doc/manual/src/notifications.md:78]: https://github.com/NixOS/hydra/blob/7b7c03a521108492ebd584ff34c8dfcac7767d15/src/script/hydra-eval-jobset#L657

File: doc/manual/src/notifications.md [hunk 3]
```diff
@@ -55,6 +55,30 @@ After processing, the row's `notificationspendingsince` column is set to null.
...
+* **When:** At the beginning of the evaluation phase for the jobset, before any work is done.
+* **Delivery Semantics:** Ephemeral. `hydra-notify` must be running to react to this event. No record of this event is stored.
+
+### `eval_added`
+
+* **Payload:** Exactly two values, separated by the two-character string `\t` (ie: not a tab): an opaque trace ID representing this evaluation, and the ID of the JobsetEval record.
+* **When:** After the evaluator fetches inputs and completes the evaluation successfully.
+* **Delivery Semantics:** Ephemeral. `hydra-notify` must be running to react to this event. No record of this event is stored.
+
+### `eval_cached`
```
Comment (grahamc) [inline doc/manual/src/notifications.md:70]: https://github.com/NixOS/hydra/blob/7b7c03a521108492ebd584ff34c8dfcac7767d15/src/script/hydra-eval-jobset#L673

File: doc/manual/src/notifications.md [hunk 4]
```diff
@@ -55,6 +55,30 @@ After processing, the row's `notificationspendingsince` column is set to null.
 
 It is possible for subsequent deliveries of the same `build_finished` data to imply different outcomes. For example, if the build fails, is restarted, and then succeeds. In this scenario the `build_finished` events will be delivered at least twice, once for the failure and then once for the success.
 
+### `eval_started`
+
+* **Payload:** Exactly three values, separated by the two-character string `\t` (ie: not a tab): an opaque trace ID representing this evaluation, the name of the project, and the name of the jobset.
+* **When:** At the beginning of the evaluation phase for the jobset, before any work is done.
+* **Delivery Semantics:** Ephemeral. `hydra-notify` must be running to react to this event. No record of this event is stored.
+
+### `eval_added`
```
Comment (grahamc) [inline doc/manual/src/notifications.md:64]: https://github.com/NixOS/hydra/blob/7b7c03a521108492ebd584ff34c8dfcac7767d15/src/script/hydra-eval-jobset#L769-L770

File: doc/manual/src/notifications.md (hunk 2 above)
Comment (grahamc) [reply doc/manual/src/notifications.md:78]: https://github.com/NixOS/hydra/blob/7b7c03a521108492ebd584ff34c8dfcac7767d15/src/script/hydra-eval-jobset#L657

**Review by grahamc — COMMENTED** [review]
Added references in the comments to where the events are sent.

File: doc/manual/src/notifications.md [hunk 5]
```diff
@@ -55,6 +55,30 @@ After processing, the row's `notificationspendingsince` column is set to null.
 
 It is possible for subsequent deliveries of the same `build_finished` data to imply different outcomes. For example, if the build fails, is restarted, and then succeeds. In this scenario the `build_finished` events will be delivered at least twice, once for the failure and then once for the success.
 
+### `eval_started`
+
+* **Payload:** Exactly three values, separated by the two-character string `\t` (ie: not a tab): an opaque trace ID representing this evaluation, the name of the project, and the name of the jobset.
```
Comment (grahamc) [inline doc/manual/src/notifications.md:60]: https://github.com/NixOS/hydra/blob/7b7c03a521108492ebd584ff34c8dfcac7767d15/src/script/hydra-eval-jobset#L863-L864

**Pull request merged**